### PR TITLE
Added TypeScript.md in Tech_Stacks

### PR DESCRIPTION
**Pull request by MehrdadQ**
Added TypeScript.md in Tech_Stacks, and updated Tech_Stacks.md to include the new addition

### CODE REVIEW

File: Topics/Tech_Stacks/TypeScript.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+## How to use TypeScript
+
+You can integrate TypeScript into any JavaScript code, but it is most commonly used for web development using JavaScript frameworks such as React and Angular.
+
+Here is how to create projects with the above frameworks using TypeScript.
+
+### React
+
+You can find the complete guide [here](https://create-react-app.dev/docs/adding-typescript/). To learn more about React and TypeScript,You can also follow [this YouTube video by Ben Awad](https://www.youtube.com/watch?v=Z5iWr6Srsj8&t=28s&ab_channel=BenAwad).
```
Comment (lmg8) [inline Topics/Tech_Stacks/TypeScript.md:27]:
```suggestion
You can find the complete guide [here](https://create-react-app.dev/docs/adding-typescript/). To learn more about React and TypeScript, you can also follow [this YouTube video by Ben Awad](https://www.youtube.com/watch?v=Z5iWr6Srsj8&t=28s&ab_channel=BenAwad).
```

**Review by lmg8 — CHANGES_REQUESTED** [review]
This is a good introduction to typescript with plenty of helpful resources. I would consider adding a table of contents similar to [MySQL page](https://github.com/learning-software-engineering/learning-software-engineering.github.io/blob/main/Topics/Tech_Stacks/Learning_MySQL.md) or [swift](https://github.com/learning-software-engineering/learning-software-engineering.github.io/blob/main/Topics/Tech_Stacks/swift.md) to make it more consistent with other tech stack pages.

Also, may you fix the merge conflicts in Tech_Stacks.md

**Review by lmg8 — APPROVED** [review]

**Review by lmg8 — CHANGES_REQUESTED** [review]
One small thing: can you update the readme.md so future contributors know where to update Typescript related resources?

**Comment by MehrdadQ** [comment]
so just add it to the main readme like that? please let me know if there is anything else

**Review by lmg8 — APPROVED** [review]

**Pull request merged**